app/views/purchases/index.html.erb

```diff
@@ -20,19 +20,22 @@
           <th>Payee</th>
           <th>Category</th>
           <th>Purchaser</th>
+          <th>Purchasee</th>
         </tr>
       </thead>
       <tbody>
 
-        <% @purchases.each do |purchase| %>
+        <% @purchases.order(:date).reverse_order.each do |purchase| %>
           <tr>
             <td><%= purchase.date.to_s %></td>
             <td><%= purchase.description %></td>
             <td><%= number_to_currency(purchase.amount) %></td>
             <td><%= purchase.payee %></td>
             <td><%= Subcategory.find(purchase[:subcategory_id]).name %></td>
-            <td><%= User.find(purchase[:user_id]).name %></td>
+            <td><%= User.find(purchase[:purchaser_id]).name %></td>
+            <td><%= User.find(purchase[:purchasee_id]).name %></td>
```

Comment:
I'm starting to wonder if it'd be better to set these as instance variables in the controller, e.g., `@subcategory`, `@purchaser`, `@purchasee`. Might pay to look up if there's a convention for this kind of thing.

Reply:
Scratch the above, but should be able to use helpers like purchase.purchasee.name in place of User.find(purchase[:purchasee_id]).name

Reply:
Great suggestion - much tidier to look at and fewer database calls.